api/types/src/transaction.rs

```diff
@@ -665,26 +668,19 @@ impl VerifyInput for TransactionPayload {
             TransactionPayload::EntryFunctionPayload(inner) => inner.verify(),
             TransactionPayload::ScriptPayload(inner) => inner.verify(),
             TransactionPayload::MultisigPayload(inner) => inner.verify(),
-            // Deprecated. Will be removed in the future.
-            TransactionPayload::ModuleBundlePayload(inner) => inner.verify(),
+
+            // Deprecated.
+            TransactionPayload::ModuleBundlePayload(_) => {
+                unreachable!("Module bundle transaction is deprecated")
```

Comment:
generally I feel uncomfortable to use unreachable in the big pr, we can return errors here and gradually migrate real unreachable cases later

Reply:
Yes, my motivation was to make error path reasoning easier (given that we gate at the entry point), but let's switch this to errors, and we can revisit for places which are truly unreachable.

Reply:
Though it is hard to make sure we have one entry point... indeed